docs/NamingConvention.md

```diff
@@ -1,37 +1,61 @@
 # Naming Convention [Proposal]
-
 ## Common naming criteria for all resources
 - Max length: `1-63` chars
 - Start with lower case character, can’t end in `-` or `_` or any special char
 > Preferred names have all lower case
 >> A preferred name can be regexed by `[a-z]([-a-z0-9]*[a-z0-9])?`
 
 ## Azure resource names
-Azure resources will have four parts:
-- `[ prefix ]-( project-short-form )-[ identifier ]-[ suffix ]-( environment-type )`
-  - Mandatory: `(project-short-form)`. Example: `ngsa`, `sp`
-  - Mandatory: `(environment-type)` - usage env. Possible val: `prod`, `test` and `dev`
-  - Optional: `[prefix]` or `[suffix]` - resource type suffix. Follows [Azure naming convention][1]
-    - Common resource prefix/suffix:
-        | Az Resource Name   | Prefix/Suffix |
-        |--------------------|----------|
-        | Key Vault          | kv-      |
-        | Function           | func-    |
-        | Virtual Machine    | vm-      |
-        | VM Scale Set       | vmss-    |
-        | CosmosDB           | cosmos-  |
-        | AKS Cluster        | aks-     |
-        | Container Registry | cr-      |
-  - Optional: `[identifier]` - refers to descriptive label
-  - Each part will be separated by a hyphen `-`
-  - Resource names will use either suffix or prefix…
-
-## Naming convention for Kubernetes resources 
-### Label format
-
-## Resourcese
+Azure resources consists of four parts: `( project-short-form )-[ identifier ]-( environment-type )-[ suffix ]`
+  - Mandatory: `(project-short-form)`.
+  - Mandatory: `(environment-type)` - usage env.
+    - Possible val: `pre`, `test`, `stage`, `prod`, and `dev` [?]
+      - `pre` is pre-production
+  - Optional: `[identifier]` - refers to descriptive label.
+  - Best effort to fit based on specific resource naming constraints.
```

Comment:
the identifier seems redundant. I can't think of a scenario where I would use it.

Reply:
I've noticed the identifier works well for resource groups, but not for individual resources. Since it is optional, it allows us to do something like `ngsa-pre-cosmos-rg` for a resource group. Then `ngsa-pre-cosmos` for cosmos itself.